blackjax/mcmc/ss.py

```diff
@@ -332,12 +317,10 @@ def sample_direction_from_covariance(
         A normalized direction vector (1D array).
     """
     p, unravel_fn = jax.flatten_util.ravel_pytree(position)
-    u = jax.random.normal(rng_key, shape=p.shape, dtype=p.dtype)
-    u /= jnp.linalg.norm(u)
-    L = jnp.linalg.cholesky(cov).astype(p.dtype)
-    dim = cov.shape[0]
-    L = L * 2 * jnp.sqrt(dim + 2)
-    d = linear_map(L, u)
+    d = jax.random.normal(rng_key, shape=p.shape, dtype=p.dtype)
+    invcov = jnp.linalg.inv(cov)
+    norm = jnp.sqrt(jnp.einsum("...i,...ij,...j", d, invcov, d))
+    d = d / norm[..., None]
```

Comment:
Here we had some conversation with @zwei-beiner about how in fact it is correct to normalise this way (non-trivially). I'm happy that this is now the correct d scaling, do we want a factor of 2 (diameter rather than radius) or something similar?